pkg/controller/storagecluster/reconcile.go

```diff
@@ -1152,3 +1156,129 @@ func getMonCount(nodeCount int) int {
 
 	return count
 }
+
+//ensureJobTemplates returns spec of a job template
+func (r *ReconcileStorageCluster) ensureJobTemplates(sc *ocsv1.StorageCluster, reqLogger logr.Logger) error {
+	osdCleanUpTemplate := &openshiftv1.Template{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      sc.Name + "-osd-removal",
+			Namespace: sc.Namespace,
+		},
+	}
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), r.client, osdCleanUpTemplate, func() error {
+		osdCleanUpTemplate.Objects = []runtime.RawExtension{
+			{
+				Object: newCleanupJob(sc),
+			},
+		}
+		osdCleanUpTemplate.Parameters = []openshiftv1.Parameter{
+			{
+				Name:     "ROOK_IMAGE",
+				Required: true,
+			},
+			{
+				Name:     "FAILED_OSD_ID",
+				Required: true,
+			},
+		}
+		return controllerutil.SetControllerReference(sc, osdCleanUpTemplate, r.scheme)
+	})
+	if err != nil && !errors.IsAlreadyExists(err) {
+		return err
```

Comment:
Add more context to the error message. Saves our debugging time later on.

Reply:
Specifically, something like `failed to create Template: %v` where you pass `err` as the variable.

Reply:
ack